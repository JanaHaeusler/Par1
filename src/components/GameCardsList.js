import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import GameCard from './GameCard'

GameCardsList.propTypes = {
    savedGameProfiles: PropTypes.array,
}

export default function GameCardsList({savedGameProfiles, onDelete}) {

    return(
    <GameCardsWrapper>
        {savedGameProfiles.map(({location, date, players, winner, shots, _id}) => 
          <GameCard key={_id} id={_id} onDelete={onDelete} location={location} date={date} players={players} winner={winner} shots={shots}/>
        )}
    </GameCardsWrapper>
    )
}

const GameCardsWrapper = styled.div`
  display: grid;
  gap: 10px;
`