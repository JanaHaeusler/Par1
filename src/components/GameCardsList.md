```js
<GameCardsList
  savedGameProfiles={[
    {
      location: 'Horner Racecourse',
      date: '11/04/2020',
      players: 'John, Jane',
      winner: 'Jane',
      shots: '89',
      _id: '1',
    },
    {
      location: 'City Park',
      date: '11/13/2020',
      players: 'Fritz, Hanni, Nanni',
      winner: 'Fritz',
      shots: '35',
      _id: '2',
    },
  ]}
  onDelete={() => {}}
/>
```
