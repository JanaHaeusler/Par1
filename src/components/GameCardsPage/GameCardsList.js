import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import GameCard from './GameCard'

GameCardsList.propTypes = {
    savedGameProfiles: PropTypes.object.isRequired,
    onDelete: PropTypes.func.isRequired,
    onEdit: PropTypes.func.isRequired,
    switchToSavedGamePage: PropTypes.func.isRequired,
}

export default function GameCardsList({savedGameProfiles, onDelete, onEdit, switchToSavedGamePage}) {

  return(
    <GameCardsWrapper>
        {savedGameProfiles.allIds.map((id) => {
          const {location, date, players, winner, shots, _id} = savedGameProfiles.byId[id]
          return <GameCard 
                    key={_id} 
                    id={_id} 
                    onDelete={onDelete} 
                    onEdit={onEdit} 
                    location={location} 
                    date={date} 
                    players={players} 
                    winner={winner} 
                    shots={shots}
                    switchToSavedGamePage={switchToSavedGamePage}/>
        })}
    </GameCardsWrapper>
  )
}

const GameCardsWrapper = styled.div`
  display: grid;
  gap: 10px;
`