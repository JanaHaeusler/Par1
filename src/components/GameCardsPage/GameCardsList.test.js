import GameCardsList from './GameCardsList'
import { render } from '@testing-library/react'

describe('GameCardsList', () => {
  
    const savedGameProfiles = {
      byId: {
        1: {
          location: 'Horner Racecourse',
          date: '2020-12-24',
          players: 'John, Jane',
          winner: 'Jane',
          shots: '89',
          _id: '1',
        },
        2: {
          location: 'City Park',
          date: '2020-01-01',
          players: 'Fritz, Hanni, Nanni',
          winner: 'Fritz',
          shots: '35',
          _id: '2',
        },
      },
      allIds: [1, 2],
    }
  
    it('renders correctly', () => {
        const onDeleteMock = jest.fn()
        const onEditMock = jest.fn()
        const { container, getByText, getAllByTestId } = render(<GameCardsList savedGameProfiles={savedGameProfiles} onDelete={onDeleteMock} onEdit={onEditMock}/>)
        
        expect(container.firstChild).toMatchSnapshot()

        expect(getByText('Horner Racecourse')).toBeInTheDocument()
        expect(getByText('2020-12-24')).toBeInTheDocument()
        expect(getByText('John, Jane')).toBeInTheDocument()
        expect(getByText('Jane')).toBeInTheDocument()
        expect(getByText('89')).toBeInTheDocument()
        
        expect(getByText('City Park')).toBeInTheDocument()
        expect(getByText('2020-01-01')).toBeInTheDocument()
        expect(getByText('Fritz, Hanni, Nanni')).toBeInTheDocument()
        expect(getByText('Fritz')).toBeInTheDocument()
        expect(getByText('35')).toBeInTheDocument()

        const ButtonDelete = getAllByTestId('button-delete-icon')
        expect(ButtonDelete).toHaveLength(2)
        const ButtonEdit = getAllByTestId('button-edit-icon')
        expect(ButtonEdit).toHaveLength(2)
      })
})