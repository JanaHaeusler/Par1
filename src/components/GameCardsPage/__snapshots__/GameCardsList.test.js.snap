// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCardsList renders correctly 1`] = `
.c1 {
  padding: 10px;
  box-shadow: 0 0 10px var(--primary-medium);
  border-radius: 10px;
  background-color: var(--primary-medium);
  font-size: 0.9rem;
}

.c2 {
  display: grid;
  grid-template-rows: repeat(4,auto);
  grid-template-columns: repeat(3,1fr);
  grid-gap: 10px;
}

.c4 {
  grid-column-start: 1;
  grid-row-start: 1;
}

.c3 {
  grid-column: 1 / 3;
  grid-row-start: 2;
  text-transform: uppercase;
}

.c5 {
  grid-column: 1 / 4;
  grid-row-start: 3;
}

.c6 {
  grid-column-start: 1;
  grid-row-start: 4;
}

.c7 {
  grid-column-start: 2;
  grid-row-start: 4;
}

.c8 {
  grid-column-start: 3;
  grid-row: 1 / 5;
  justify-self: end;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 5px;
  padding: 0;
  width: 30px;
  border: none;
  background: none;
}

.c10 {
  margin-right: 3px;
  fill: var(--secondary-dark);
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 20px 5px 5px 5px;
  padding: 0;
  width: 30px;
  border: none;
  background: none;
}

.c12 {
  margin-right: 3px;
  fill: var(--secondary-dark);
}

.c0 {
  display: grid;
  gap: 10px;
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        Horner Racecourse
      </div>
      <div
        class="c4"
      >
        2020-12-24
      </div>
      <div
        class="c5"
      >
        <h4>
          Player(s)
        </h4>
        <span>
          John, Jane
        </span>
      </div>
      <div
        class="c6"
      >
        <h4>
          Winner(s)
        </h4>
        <span>
          Jane
        </span>
      </div>
      <div
        class="c7"
      >
        <h4>
          Total Shots
        </h4>
        <span>
          89
        </span>
      </div>
      <div
        class="c8"
      >
        <button
          class="c9"
          data-testid="button-delete-icon"
        >
          <svg
            class="c10"
          >
            bin.svg
          </svg>
        </button>
        <button
          class="c11"
          data-testid="button-edit-icon"
        >
          <svg
            class="c12"
          >
            pencil.svg
          </svg>
        </button>
      </div>
    </div>
  </section>
  <section
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        City Park
      </div>
      <div
        class="c4"
      >
        2020-01-01
      </div>
      <div
        class="c5"
      >
        <h4>
          Player(s)
        </h4>
        <span>
          Fritz, Hanni, Nanni
        </span>
      </div>
      <div
        class="c6"
      >
        <h4>
          Winner(s)
        </h4>
        <span>
          Fritz
        </span>
      </div>
      <div
        class="c7"
      >
        <h4>
          Total Shots
        </h4>
        <span>
          35
        </span>
      </div>
      <div
        class="c8"
      >
        <button
          class="c9"
          data-testid="button-delete-icon"
        >
          <svg
            class="c10"
          >
            bin.svg
          </svg>
        </button>
        <button
          class="c11"
          data-testid="button-edit-icon"
        >
          <svg
            class="c12"
          >
            pencil.svg
          </svg>
        </button>
      </div>
    </div>
  </section>
</div>
`;
